### DISCUSSION TITLE
Route manifest (or named route functions)

### DISCUSSION THREAD

**Discussion by frontsideair** [remix-run/remix#2797] [Proposals]
Full stack React frameworks such as [Blitz.js](https://blitzjs.com) and [RedwoodJS](https://redwoodjs.com) share a `Link` component that is aware of the routes that your app has. This way, creating links is safer and you can use intellisense instead of randomly typing strings.

Blitz.js calls this [route manifest](https://blitzjs.com/docs/route-manifest) and in RedwoodJS it's [named route functions](https://redwoodjs.com/docs/router#link-and-named-route-functions).

Implementation-wise, Blitz.js generates the manifest using CLI commands, and I assume RedwoosJS does something similar. Since Remix gives so much safety, adding this would be in line with the project's vision and provide a much better developer experience.

**Comment by frontsideair** [comment]
Just found out about [routes-gen](https://github.com/sandulat/routes-gen), which solves my issue but would love to see it integrated into Remix.

**Reply by nickluger** [reply]
Thanks for sharing `routes-gen`. 💙  Would love to see it integrated in Remix, too.

**Reply by frontsideair** [reply]
There’s also https://github.com/yesmeck/remix-routes now.

**Comment by lpsinger** [comment]
Possibly also related: #2912.
